src/models/user.js

```diff
@@ -0,0 +1,19 @@
+import { Schema, model } from "mongoose";
+
+const schema = new Schema({
+  name: { type: String, default: null, required: false },
+  preferred_mode: {
+    type: String,
+    enum: ["phone", "earbuds"],
+    default: "phone",
+    required: true,
+  },
+  daily_goal: { type: Number, min: 50, default: 50, required: true },
+  level_id: { type: Schema.Types.ObjectId, ref: "Level", required: false }, //TODO: Update to "true" when Level collection is populated
+  is_setup_complete: { type: Boolean, default: false, required: true },
+  xp: { type: Number, min: 0, default: 0, required: true },
+  hp: { type: Number, min: 0, default: 100, required: true },
+  device_id: { type: String, unique: true, required: true },
```

Comment:
As a side note, we could leave the device_id as not required and default null for now.

Expo gives us a unique identifier for android but not for ios
https://docs.expo.dev/versions/latest/sdk/device/
<img width="1101" alt="Screenshot 2024-06-12 at 11 09 04 AM" src="https://github.com/LangaraBeetles/weaup-api/assets/57273227/336664dd-366c-4f34-8981-bedf77f7f91f">

Reply:
This is done